commands/run/index.js

```diff
@@ -99,21 +99,41 @@ class RunCommand extends Command {
           const exitCode = Math.max(...codes, 1);
 
           this.logger.error("", "Received non-zero exit code %d during execution", exitCode);
+          if (!this.options.stream) {
+            results
+              .filter((result) => result.failed)
+              .forEach((result) => {
+                this.logger.error("", result.pkg.name, result.stderr);
+              });
+          }
           process.exitCode = exitCode;
         }
+        return results;
       });
     }
 
-    return chain.then(() => {
-      this.logger.success(
+    return chain.then((results) => {
+      const someFailed = results.some((result) => result.failed);
+
+      (someFailed ? this.logger.error : this.logger.success)(
         "run",
         "Ran npm script '%s' in %d %s in %ss:",
         this.script,
         this.count,
         this.packagePlural,
         (getElapsed() / 1000).toFixed(1)
       );
-      this.logger.success("", this.packagesWithScript.map((pkg) => `- ${pkg.name}`).join("\n"));
+      if (!this.bail) {
```

Comment:
this line should be 
```ts
if (!this.bail && !this.options.stream) {
```
since that was mentioned in the PR that it only works when it's not streaming, without this `result.pkg` (called with the `this.logger`) is undefined and calling `result.pkg.name` throws an error